### PR TITLE
TXAD-89: Implement Get and Post API for Checklist System User.

### PR DESCRIPTION
**Pull request by ruchawajpe**

### CODE REVIEW

File: lib/blocs/checklist/systemUser/pdf/sys_user_pdf_bloc.dart [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+  }
+
+  FutureOr<void> _fetchPdf(FetchPdf event, Emitter<FetchPdfStates> emit) async {
+    emit(FetchingPdf());
+    try {
+      String hashCode = (await _customerCache.getHashCode(CacheKeys.hashcode))!;
+      GetPdfModel getPdfModel = await _sysUserCheckListRepository.fetchPdf(
+          event.responseId, hashCode);
+      var decrypted = PdfEncrypt.decryptAES(getPdfModel.message,
+          "SvwH32gnWK1slqvskIsSg9duoVfgOQLWitcfZGr+n2KX1yltKm2T+EbsIhBm0E6B");
```
Comment (Aditi-005) [inline lib/blocs/checklist/systemUser/pdf/sys_user_pdf_bloc.dart:30]: Get this from shared prefs

File: lib/screens/checklist/systemUser/widgets/sys_user_list_section.dart [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+      BlocConsumer<SysUserCheckListBloc, SysUserCheckListStates>(
+          listenWhen: (previousState, currentState) =>
+              currentState is FetchingCheckList && checklistData.isNotEmpty ||
+              currentState is CheckListError && checklistData.isNotEmpty,
+          buildWhen: (previousState, currentState) =>
+              currentState is CheckListFetched ||
+              currentState is CheckListError && checklistData.isEmpty,
+          listener: (context, state) {
+            if (state is FetchingCheckList && checklistData.isNotEmpty) {
+              showCustomSnackBar(context, 'Load more...', '');
```
Comment (Aditi-005) [inline lib/screens/checklist/systemUser/widgets/sys_user_list_section.dart:37]: Add in string constants

File: lib/screens/checklist/systemUser/widgets/sys_user_list_section.dart [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+            } else if (state is CheckListError &&
+                checklistData.isNotEmpty &&
+                context.read<SysUserCheckListBloc>().filterData == '{}') {
+              showCustomSnackBar(context, 'No more data!', '');
```
Comment (Aditi-005) [inline lib/screens/checklist/systemUser/widgets/sys_user_list_section.dart:41]: Add in string constants

File: lib/utils/pdf_encrypt_data.dart [hunk 4]
```diff
@@ -0,0 +1,20 @@
+import 'package:encrypt/encrypt.dart';
+
```
Comment (Aditi-005) [inline lib/utils/pdf_encrypt_data.dart:2]: Remove file, this function is done in encrypt file already

**Review by Aditi-005 — CHANGES_REQUESTED** [review]
Please do the changes.

File: lib/blocs/checklist/systemUser/approvePopUp/sys_user_approve_pop_up_bloc.dart [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+import 'package:flutter_bloc/flutter_bloc.dart';
+import 'package:toolkit/blocs/checklist/systemUser/approvePopUp/sys_user_approve_pop_up_events.dart';
+import 'package:toolkit/blocs/checklist/systemUser/approvePopUp/sys_user_approve_pop_up_states.dart';
+import '../../../../data/cache/cache_keys.dart';
+import '../../../../data/cache/customer_cache.dart';
+import '../../../../data/models/checklist/systemUser/sys_user_approve_model.dart';
+import '../../../../di/app_module.dart';
+import '../../../../repositories/checklist/systemUser/sys_user_checklist_repository.dart';
+
+class ApproveBloc extends Bloc<ApproveEvent, ApproveStates> {
```
[the remark this answers is not in the document]
Comment (ruchawajpe) [reply lib/blocs/checklist/systemUser/approvePopUp/sys_user_approve_pop_up_bloc.dart:11]: CheckList

File: lib/blocs/checklist/systemUser/pdf/sys_user_checklist_pdf_events.dart [hunk 6]
```diff
@@ -0,0 +1,5 @@
+class FetchCheckListPdf {
```
Comment (Aditi-005) [inline lib/blocs/checklist/systemUser/pdf/sys_user_checklist_pdf_events.dart:1]: FetchCheckListPdf to FetchCheckListPdfEvent

File: lib/blocs/checklist/systemUser/scheduleDatesResponse/checklist_schedule_dates_response_bloc.dart [hunk 7]
```diff
@@ -0,0 +1,101 @@
...
+    emit(CheckListWorkforceListFetched(
+        checkListWorkforceListModel: event.checkListWorkforceListModel,
+        selectedIResponseIdList: responsesIdsList,
+        popUpMenuBuilder: event.popUpBuilder));
+  }
+
+  _fetchPopUpMenuItems(FetchCheckListPopUpMenu event,
+      Emitter<CheckListScheduleDatesResponseStates> emit) {
+    List popUpMenuItems = [
+      'Approve',
```
Comment (Aditi-005) [inline lib/blocs/checklist/systemUser/scheduleDatesResponse/checklist_schedule_dates_response_bloc.dart:89]: User database keys

File: lib/screens/checklist/systemUser/sys_user_pop_up_menu_screen.dart [hunk 8]
```diff
@@ -0,0 +1,102 @@
...
+          if (state is CheckListPopUpMenuItemsFetched) {
+            return PopupMenuButton(
+                shape: RoundedRectangleBorder(
+                  borderRadius: BorderRadius.circular(kCardRadius),
+                ),
+                iconSize: kIconSize,
+                icon: const Icon(Icons.more_vert_outlined),
+                offset: const Offset(0, xxTinierSpacing),
+                onSelected: (value) {
+                  if (value == 'Approve') {
```
Comment (Aditi-005) [inline lib/screens/checklist/systemUser/sys_user_pop_up_menu_screen.dart:54]: user database keys

File: lib/screens/checklist/systemUser/widgets/sys_user_list_section.dart [hunk 9]
```diff
@@ -0,0 +1,135 @@
...
+                                    style: Theme.of(context)
+                                        .textTheme
+                                        .small
+                                        .copyWith(color: AppColor.black)),
+                              ),
+                              subtitle: Column(
+                                  crossAxisAlignment: CrossAxisAlignment.start,
+                                  children: [
+                                    Text(
+                                        '${checklistData[index].categoryname} -- ${checklistData[index].subcategoryname.toString()}',
```
Comment (Aditi-005) [inline lib/screens/checklist/systemUser/widgets/sys_user_list_section.dart:92]: remove .toString()

**Review by Aditi-005 — APPROVED** [review]
Approved!

**Pull request merged**